tools/autograd/derivatives.yaml

```diff
@@ -1273,6 +1273,12 @@
   source: grad.take(index).reshape_as(source)
   result: auto_linear  # It is affine, but sure
 
+- name: put(Tensor self, Tensor index, Tensor source, bool accumulate=False) -> Tensor
+  self: "accumulate ? grad : grad.put(index, zeros_like(source), false)"
+  index: non_differentiable
+  source: grad.take(index).reshape_as(source)
+  result: auto_linear  # It is affine, but sure
```

Comment:
cc @soulitzer @albanD do we still need the gradient formulas for put_? They look exactly the same as the one for put but I'm not sure how in-place operations get gradient formulas generated for them

Reply:
Not any more, good catch! However, to avoid a perf regression we'll want to replace `auto_linear` (which turns into `self_t.copy_(at::put(...))`) with `self_t.put(index, source_t, accumulate)` which generates `self_t.put_(...)`. See https://github.com/pytorch/pytorch/pull/74017.